### PR TITLE
HTTP Client examples

### PR DESCRIPTION
**Pull request by fwilhe**
Add sample code for HTTP Client, as supplemental material for the docs. This code is intended to help users get started and play with the client. It includes an express-based server for easy testing.

### CODE REVIEW

File: samples/http-client-examples/client.mts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  HttpResponse,
+  ParameterEncoder,
+} from "@sap-cloud-sdk/http-client";
+import { setGlobalLogLevel } from "@sap-cloud-sdk/util";
+
+// Change to 'debug' to see more about http client internals
+setGlobalLogLevel("info");
+
+function printResponse(r: HttpResponse) {
+  console.log(
```
Comment (FrankEssenberger) [inline samples/http-client-examples/client.mts:13]: [pp] Use the SDK logger?

File: samples/http-client-examples/client.mts [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import {
```
Comment (FrankEssenberger) [inline samples/http-client-examples/client.mts:1]: [req] I think we must ensure that the samples are not degrading over time. Could you perhaps add a test which starts the server and executes the clients. No big assertions just that the server and client start.

File: samples/http-client-examples/server.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const express = require("express");
+const app = express();
+const port = 3000;
+
+app.head("/csrf-token", async (req, res) => {
```
Comment (FrankEssenberger) [inline samples/http-client-examples/server.js:5]: [pp]I would assume there are also `@types/express` so you could make this TS and get some type safety.

**Comment by fwilhe** [comment]
Used a workaround for running the tests in actions because running it as part of the matrix caused issues because both instances used the same port. For the sake of simplicity I chose not to make the port configurable but to have two separate jobs which run in isolation.

Addressed the other comments, hope that's ok @FrankEssenberger

File: samples/http-client-examples/server.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    res.send("Request without token");
+  }
+});
+
+app.get("/encoding", async (req, res) => {
+  res.send(req.url);
+});
+
+app.get("/origin", async (req, res) => {
+  const result = req.headers;
```
[the remark this answers is not in the document]
Comment (fwilhe) [reply samples/http-client-examples/server.js]: which parameters do you mean? url encoded parameters are in the url, and request headers are also included.

**Pull request merged**